### PR TITLE
test/k8s: Migrate L7 visibility tests to hubble

### LINKED ISSUE #11396: CI/Hubble: Test that HTTP/DNS L7 events are stored as expected
Replace the use of Cilium Monitor with `hubble observe --follow`:

https://github.com/cilium/cilium/blob/2ff7193bb2301523af39695646ac4eb2a1294f17/test/k8sT/Policies.go#L972-L978

Current test looks for:
```
reStr = "Request dns from.*Forwarded DNS Query:.*"
```

```
reStr = fmt.Sprintf("verdict Forwarded GET http://%s/public", resource)
```

### PR DESCRIPTION
**Pull request by glibsm**
Remove usage of monitor in the L7 visibility tests and
replace them with hubble using jsonpath filter to validate
concrete items.

Fixes #11396

Signed-off-by: Glib Smaga <code@gsmaga.com>

### CODE REVIEW

**Review by michi-covalent — APPROVED** [review]
neat

**Comment by tklauser** [comment]
test-me-please

**Comment by aanm** [comment]
@glibsm I think the tests failures are related with this change.

File: test/k8sT/Policies.go [hunk 1]
```diff
@@ -98,6 +98,9 @@ var _ = Describe("K8sPolicyTest", func() {
 		daemonCfg = map[string]string{
 			"global.tls.secretsBackend": "k8s",
 			"global.debug.verbose":      "flow",
+			// enable hubble server and the CLI client
+			"global.hubble.enabled":     "true",
+			"global.hubble.cli.enabled": "true",
```
Comment (gandro) [inline test/k8sT/Policies.go:103]: Note: You will have to cleanup the Hubble Client pods spawned by this in `AfterAll`. See https://github.com/cilium/cilium/pull/11687

**Comment by glibsm** [comment]
> @glibsm I think the tests failures are related with this change.

Can you point to the failure? I can only find an unrelated hubble test breaking which is being tracked elsewhere

**Review by gandro — CHANGES_REQUESTED** [review]
#11687 has been merged. Please insert a call to `kubectl.DeleteHubbleClientPods(helpers.CiliumNamespace)` in `AfterAll`. Otherwise this test will leave hubble-cli pods running after the test has finished.

**Comment by glibsm** [comment]
> #11687 has been merged. Please insert a call to `kubectl.DeleteHubbleClientPods(helpers.CiliumNamespace)` in `AfterAll`. Otherwise this test will leave hubble-cli pods running after the test has finished.

Will rebase, thanks.

File: test/k8sT/Policies.go [hunk 2]
```diff
@@ -1075,6 +1104,8 @@ var _ = Describe("K8sPolicyTest", func() {
 			// avoid incomplete teardown if any step fails.
 			_ = kubectl.Delete(demoYAML)
 			ExpectAllPodsTerminated(kubectl)
+			kubectl.DeleteHubbleClientPods(helpers.CiliumNamespace)
+			kubectl.DeleteHubbleRelay(helpers.CiliumNamespace)
```
Comment (gandro) [inline test/k8sT/Policies.go]: hubble-relay does not need to be removed, since you are not deploying it

Comment (glibsm) [reply test/k8sT/Policies.go]: fair.

**Review by gandro — APPROVED** [review]

**Comment by gandro** [comment]
test-me-please

**Pull request merged**